lib/practice_exercises.rb

```diff
@@ -1,13 +1,45 @@
-
-# Time Complexity: ?
-# Space Complexity: ?
+require 'pry'
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  index = 0
+  while index <= list.length
+    if list[index] == list[index + 1]
+      list.delete_at(index)
```

Comment:
`delete_at` shifts each subsequent element left by one index.  So that's an O(n) method.  Since you have it nested in a loop running n times, `remove_duplicates` becomes an O(n<sup>2</sup>) method.